jsEngine/settings/Settings.ts

```diff
@@ -20,18 +21,92 @@ export class JsEnginePluginSettingTab extends PluginSettingTab {
 	}
 
 	display(): void {
-		this.containerEl.empty();
+		const containerEl = this.containerEl;
+		const settings = this.plugin.settings;
+		containerEl.empty();
 
-		if (!this.plugin.settings) {
+		if (!settings) {
 			return;
 		}
 
-		// this.containerEl.createEl('p', { text: 'Currently Empty, but there will be stuff here later.' });
+		new Setting(containerEl)
+			.setName('JS snippets (loaded on startup)')
+			.setHeading()
+			.addExtraButton(el => {
+				el.setTooltip('Reload snippets')
+					.setIcon('refresh-cw')
+					.onClick(() => this.display());
+			})
+			.addExtraButton(el => {
+				el.setTooltip('Open snippets folder')
+					.setIcon('folder-open')
+					.onClick(async () => await this.openStartupScriptsDirectory());
+			});
 
-		new Setting(this.containerEl).setName('Startup scripts').addButton(button => {
-			button.setButtonText('Manage').onClick(() => {
-				new StartupScriptsModal(this.plugin).open();
+		new Setting(containerEl)
+			.setName('Custom JS Snippets Folder')
+			.setDesc('The folder to search for JavaScript files to load')
+			.addText(el => {
+				el.setPlaceholder(this.getDefaultStartupScriptsDirectory())
+					.setValue(settings.startupScriptsDirectory ?? '')
+					.onChange(async (val: string) => {
+						if (val) {
+							settings.startupScriptsDirectory = normalizePath(val);
+						} else {
+							settings.startupScriptsDirectory = undefined;
+						}
+						await this.plugin.saveSettings();
```

Comment:
Doesn't the settings tab need to be rendered after a change to scripts folder?

Reply:
Good question, should this happen automatically or require a click onto the reload button.

I will look into that in combination with the hole removal/clearing uppon changing the directory.

Reply:
I feel like this should happen automatically. But the old enabled scripts should stay enabled.

Reply:
Hm, there is a problem. `onChange()` triggers for every keypress and rerendering results in focus-loss on the inputfield, which is very anoying.

I will try to implement re-rendering of only the subcomponents for the toggles after the rest is working.